server/src/test/java/org/opensearch/cluster/routing/allocation/RemoteStoreMigrationAllocationDeciderTests.java

```diff
@@ -0,0 +1,964 @@
+
+        ShardRouting primaryShardRouting = clusterState.getRoutingTable().shardRoutingTable(TEST_INDEX, 0).primaryShard();
+        RoutingNode nonRemoteRoutingNode = clusterState.getRoutingNodes().node(nonRemoteNode.getId());
+
+        RemoteStoreMigrationAllocationDecider remoteStoreMigrationAllocationDecider = new RemoteStoreMigrationAllocationDecider(
+            customSettings,
+            getClusterSettings(customSettings)
+        );
+
+        RoutingAllocation routingAllocation = new RoutingAllocation(
+            new AllocationDeciders(Collections.singleton(remoteStoreMigrationAllocationDecider)),
+            clusterState.getRoutingNodes(),
+            clusterState,
+            null,
+            null,
+            0L
+        );
+        routingAllocation.debugDecision(true);
+
+        Decision decision = remoteStoreMigrationAllocationDecider.canAllocate(primaryShardRouting, nonRemoteRoutingNode, routingAllocation);
+        assertThat(decision.type(), is(Decision.Type.YES));
+        assertThat(
+            decision.getExplanation().toLowerCase(Locale.ROOT),
+            is("[remote_store migration_direction]: primary shard copy can be allocated to a non-remote node for strict compatibility mode")
+        );
+
+        indexMetadataBuilder = getIndexMetadataBuilder(true, 1, 0);
+        customSettings = getCustomSettings(REMOTE_STORE_DIRECTION, STRICT_MODE, indexMetadataBuilder);
+
+        DiscoveryNode remoteNode = getRemoteNode();
+        assertTrue(remoteNode.isRemoteStoreNode());
+
+        discoveryNodes = DiscoveryNodes.builder().add(remoteNode).localNodeId(remoteNode.getId()).build();
+
+        clusterState = getInitialClusterState(customSettings, indexMetadataBuilder, discoveryNodes);
+
+        primaryShardRouting = clusterState.getRoutingTable().shardRoutingTable(TEST_INDEX, 0).primaryShard();
+        RoutingNode remoteRoutingNode = clusterState.getRoutingNodes().node(remoteNode.getId());
+
+        remoteStoreMigrationAllocationDecider = new RemoteStoreMigrationAllocationDecider(
+            customSettings,
+            getClusterSettings(customSettings)
+        );
+
+        routingAllocation = new RoutingAllocation(
+            new AllocationDeciders(Collections.singleton(remoteStoreMigrationAllocationDecider)),
+            clusterState.getRoutingNodes(),
+            clusterState,
+            null,
+            null,
+            0L
+        );
+        routingAllocation.debugDecision(true);
+
+        decision = remoteStoreMigrationAllocationDecider.canAllocate(primaryShardRouting, remoteRoutingNode, routingAllocation);
+        assertThat(decision.type(), is(Decision.Type.YES));
+        assertThat(
+            decision.getExplanation().toLowerCase(Locale.ROOT),
+            is("[remote_store migration_direction]: primary shard copy can be allocated to a remote node for strict compatibility mode")
+        );
+    }
+
+    public void testAlwaysAllocateNewReplicaShardForStrictMode() {
```

Comment:
lets combine primary and replica into one test and chose primary/replica randomly .